src/compiler/checker.ts

```diff
@@ -14054,7 +14054,10 @@ namespace ts {
             }
 
             if (isHeritageClauseElementIdentifier(<EntityName>entityName)) {
-                let meaning = entityName.parent.kind === SyntaxKind.ExpressionWithTypeArguments ? SymbolFlags.Type : SymbolFlags.Namespace;
+                let meaning = entityName.parent.kind === SyntaxKind.ExpressionWithTypeArguments ?

```

Comment:
You should only add the `Value` flag if this is a heritage clause of a `class` expression (see other comment)